File: starostina/pattern_matching/brute_force.py
def brute_force(text, pattern):
	res = []
	text_len = len(text)
	pattern_len = len(pattern)

	for i in xrange(text_len - pattern_len + 1):
		if (all(text[i + j] == pattern[j] for j in xrange(pattern_len))):
			res.append(i)
	return res

if (__name__ == "__main__"):
	text = raw_input()
	pattern = raw_input()
	print brute_force(text, pattern)

File: lipovich/task#1/bruteforce.cpp
#include "bruteforce.h"

TResult bruteforce(const std::string& p, const std::string& t)
{
    int n = p.size(), m = t.size();

    TResult res;

    for (int i = 0; i < m - n + 1; ++i)
    {
       int j = 0;

       while (j < n && p[j] == t[i + j])
       {
           ++j;
       }

       if (j == n)
       {
            res.push_back(i);
            std::cout << i << std::endl;
       }
    }

    return res;
}
